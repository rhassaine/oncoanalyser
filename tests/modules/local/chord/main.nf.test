nextflow_process {
    name "Test Process CHORD"
    script "modules/local/chord/main.nf"
    process "CHORD"
    config "tests/nextflow.config"

    test("CHORD with default parameters") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                mode = 'WGS'
            }
            process {
                """
                input[0] = [
                    [id: 'test', sample_id: 'subject_a.tumor'],
                    file('/home/rhassaine/container_test_output/subject_a/purple/subject_a.tumor.purple.somatic.vcf.gz', checkIfExists: true),
                    file('/home/rhassaine/container_test_output/subject_a/esvee/subject_a.tumor.esvee.somatic.vcf.gz', checkIfExists: true)
                ]
                input[1] = file(params.resource_path + "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna", checkIfExists: true)
                input[2] = file(params.resource_path + "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.fai", checkIfExists: true)
                input[3] = file(params.resource_path + "GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.dict", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.chord_dir != null },
                { assert process.out.chord_dir.size() > 0 },
                { assert snapshot(process.out).match() }
            )

            if (process.success) {
                def chord_output_dir = process.out.chord_dir[0][1]
                def sample_id = process.out.chord_dir[0][0].sample_id
                assertAll(
                    { assert path(chord_output_dir).resolve("${sample_id}.chord.mutation_contexts.tsv").exists() },
                    { assert path(chord_output_dir).resolve("${sample_id}.chord.prediction.tsv").exists() },
                    { assert process.out.versions != null },
                    { assert process.out.versions.size() > 0 },
                    { assert path(process.out.versions[0]).exists() }
                )
            }
        }
    }

    test("CHORD should fail when input is missing") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                mode = 'WGS'
            }
            process {
                """
                input[0] = [
                    [id: 'test', sample_id: 'subject_a.tumor'],
                    [],  // Missing SMLV VCF file
                    []   // Missing SV VCF file
                ]
                """
            }
        }

        then {
            assertAll(
                { assert !process.success },
                { assert process.exitStatus != 0 }
            )
        }
    }
}