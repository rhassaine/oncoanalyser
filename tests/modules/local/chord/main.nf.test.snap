{
    "CHORD with default parameters": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "sample_id": "subject_a.tumor"
                        },
                        [
                            "subject_a.tumor.chord.mutation_contexts.tsv:md5,3012b46fa2dd1f76225bfc825e5a40a6",
                            "subject_a.tumor.chord.prediction.tsv:md5,9c0971533aca96471dc7c9dac637ce0e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,515be517b16e231349d32b81d67174ae"
                ],
                "chord_dir": [
                    [
                        {
                            "id": "test",
                            "sample_id": "subject_a.tumor"
                        },
                        [
                            "subject_a.tumor.chord.mutation_contexts.tsv:md5,3012b46fa2dd1f76225bfc825e5a40a6",
                            "subject_a.tumor.chord.prediction.tsv:md5,9c0971533aca96471dc7c9dac637ce0e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,515be517b16e231349d32b81d67174ae"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-20T03:37:08.865934053"
    }
}